src/main/java/com/conveyal/analysis/models/AnalysisRequest.java

```diff
@@ -150,20 +152,20 @@
     public DecayFunction decayFunction;
 
     /**
-     * Get all of the modifications for a project id that are in the Variant and map them to their
-     * corresponding r5 mod
+     * Create the R5 `Scenario` from this request.
      */
-    private static List<Modification> modificationsForProject (
-            UserPermissions userPermissions,
-            String projectId,
-            int variantIndex)
-    {
-        return Persistence.modifications
-                .findPermitted(QueryBuilder.start("projectId").is(projectId).get(), userPermissions)
-                .stream()
-                .filter(m -> variantIndex < m.variants.length && m.variants[variantIndex])
-                .map(com.conveyal.analysis.models.Modification::toR5)
-                .collect(Collectors.toList());
+    public Scenario createScenario (UserPermissions userPermissions) {
+        QueryBuilder query = "all".equals(scenarioId)
+                ? QueryBuilder.start("projectId").is(projectId)
+                : QueryBuilder.start("_id").in(modificationIds);
```

Comment:
What is the special significance of scenarioId == "all"?

Reply:
It's the ID that the UI sends for the "All modifications" scenario. Which includes all modifications within a project.

Reply:
Clarified in Slack chat: we now have two implicit scenarios, "baseline" and "all". The "all" scenario always contains every modification for a project, even as that list grows or shrinks. Theoretically there's a potential timing issue here where the backend's view of the database is out of sync with the UI's, where a modification is added or changed in the database between the time the UI saw it and the backend receives this request. We will think through ways to mitigate these risks, but for the time being the use of "all" is certainly no worse than existing mechanisms.